File: python_solution/problem3.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Fri Aug 23 20:36:12 2024

@author: sharminalina
"""
class Problem3:
    def lengthOfLongestSubstring(self, s: str) -> int:
        substrings = set()
        start = 0
        max_length = 0

        for i in range(len(s)):
            while s[i] in substrings:
                substrings.remove(s[start])
                start += 1
            substrings.add(s[i])
            max_length = max(max_length, i - start + 1)

        return max_length

def main():
    # Read input from the console
    s = input("Enter a string: ")

    # Create an instance of Solution and find the length of the longest substring
    solution_prob3 = Problem3()
    result = solution_prob3.lengthOfLongestSubstring(s)

    # Print the result
    print(f"Length of the longest substring without repeating characters: {result}")

if __name__ == "__main__":
    main()

File: C#Solutions/problem3.cs
using System;
using System.Collections.Generic;

namespace AllProblems
{
    public static class Problem3
    {
        public static void Solve()
        {
            Console.WriteLine("Solution to Problem 1");
            // Add the problem 1 logic here
            Console.WriteLine("Enter a string to find the length of the longest substring without repeating characters:");
            string s = Console.ReadLine();
            int result = LengthOfLongestSubstring(s);
            Console.WriteLine(result);
        }

        public static int LengthOfLongestSubstring(string s) {
            HashSet<char> substrings = new HashSet<char>();

            int start = 0;
            int max_length = 0;

            for (int i = 0; i < s.Length; i++){
                while(substrings.Contains(s[i])){
                    substrings.Remove(s[start]);
                    start++;
                }
                substrings.Add(s[i]);
                max_length = Math.Max(max_length, i - start +1);

            }
            return max_length;
        
        }
    }
}